plugins/cortex_v2/icon_cortex_v2/util/api.py

```diff
@@ -0,0 +1,122 @@
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
+from json.decoder import JSONDecodeError
+from requests import request
+from requests.exceptions import ConnectionError
+
+# Custom imports below
+from icon_cortex_v2.util.util import eq_
+
+
+class API:
+    def __init__(self, url: str, api_key: str, verify_cert: bool, proxies: dict):
+        self.base_url = f"{url}/api"
+        self.api_key = api_key
+        self.verify_cert = verify_cert
+        self.proxies = proxies
+
+    def send_request(self, method: str, path: str, data: dict = None, params: dict = None, files: dict = None):
+        method = method.upper()
+        headers = {"Authorization": f"Bearer {self.api_key}"}
```

Comment:
according to that [documentation](https://github.com/TheHive-Project/CortexDocs/blob/master/api/api-guide.md#introduction) basic auth is also supported. Should we support it?

Reply:
Outside of scope for this ticket. We aim to maintain the current functionality of the plugin without the external dependency, which does not support basic auth,